impl/src/main/java/com/sun/faces/lifecycle/TokenGenerator.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.sun.faces.lifecycle;
+
+import java.math.BigInteger;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ *
+ * @author lu4242
+ */
+class TokenGenerator
+{
+    private final AtomicLong seed;
+
+    public TokenGenerator()
+    {
+        seed = new AtomicLong(generateSeed());
+    }
+
+    private long generateSeed()
+    {
+        SecureRandom rng;
+        try
+        {
+            // try SHA1 first
+            rng = SecureRandom.getInstance("SHA1PRNG");
```

Comment:
Would it make more sense here to just call getInstanceStrong so we use what is currently considered the strong instance based on whatever jvm is running?

Reply:
I can make that change. As I noted above, I just copied their code over but have not examined it for optimizations (assumptions ftw, I guess :). I've made the change.